src/main/java/Duke.java

```diff
@@ -1,10 +1,291 @@
+                    listCount++;
+                    printTask(todoTask, listCount);
+                    writeData();
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+                    System.out.println("Please format your input as 'todo <task>'");
+                    System.out.println(LINE);
+                }
+
+            } else if (line.startsWith("deadline")) {
+                try {
+                    Deadline deadlineTask = deadline(line);
+                    listCount++;
+                    printTask(deadlineTask, listCount);
+                    writeData();
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("☹ OOPS!!! The format of deadline is wrong");
+                    System.out.println("Please format your input as 'deadline <task>/<due date>'");
+                    System.out.println(LINE);
+                }
+
+            } else if (line.startsWith("event")) {
+                try {
+                    Event eventTask = event(line);
+                    listCount++;
+                    printTask(eventTask, listCount);
+                    writeData();
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("☹ OOPS!!! The format of event is wrong");
+                    System.out.println("Please format your input as 'event <task>/<event date and time>'");
+                    System.out.println(LINE);
+                }
+
+            } else if (line.startsWith("delete")) {
+                try {
+                    delete(line);
+                    writeData();
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("☹ OOPS!!! The task number is not valid");
+                    System.out.println("Please check again and format your input as 'delete <task number>'");
+                    System.out.println(LINE);
+                }
+
+            } else {
+                throw new InvalidCommandException();
+            }
+        } catch (InvalidCommandException e) {
+            System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            System.out.println(LINE);
+        }
+    }
+
+    public static void createFile() {
+        File f = new File(DATA_FILE_PATH);
+        try {
+            f.getParentFile().mkdirs();
+            if (!f.exists()) {
+                f.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println(" I/O ERROR ");
+        }
+    }
+
+    public static void readData() throws FileNotFoundException {
+        File f = new File(DATA_FILE_PATH);
+        createFile();
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String line = s.nextLine();
+
+            if (line.startsWith("T")) {
```

Comment:
Perhaps you can try to [avoid code duplication](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#minimise-code-duplication) here as well.